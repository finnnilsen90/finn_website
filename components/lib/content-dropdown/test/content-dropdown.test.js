const React = require('react');
const ReactTestUtils = require('react-dom/test-utils');
const renderer = require('react-test-renderer')
let { shallow, mount, render } = require('enzyme');

const Content = require('../content-dropdown.js');

let newContent = new Content();

describe('component connection', () => {

    it('should be defined', () => {
        expect(Content).toBeDefined();
    })

})  
    
describe('<Content /> render', () => {

    const wrapper = render(<Content />);

    it('renders components', () => {
        expect(wrapper).toMatchSnapshot();
    });


})

describe('<Content /> shallow', () => {

    const wrapper = shallow(<Content title='test' class='test' content='test content' />, { attachTo: document.body });
    
    it('should match props', () => {
        expect(wrapper).toMatchSnapshot();
    });
        
})

describe('<Content /> mount', () => {

    const wrapper = shallow(<Content title='test' class='test' content='test content' />, { attachTo: document.body });
    const arrow = wrapper.find('.arrow');
    
    it('should match props', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('state should be false on load then true on click', () => {
        expect(wrapper.state().arrow).toEqual(false);
        arrow.simulate('click');
        expect(wrapper.state().arrow).toEqual(true);
    })

    // it('state should be true', () => {
        
    //     expect(wrapper).toMatchSnapshot();
    // })
        
})

describe('random class should be generated', () => {

    let wrapper = mount(<Content class='test'/>, { attachTo: document.body })
    let arw = wrapper.state().arw;
    let arw_click = wrapper.state().arw_click;
    let cont_par = wrapper.state().cont_par;
    let cont_hr = wrapper.state().cont_hr;

    it('should match assigned class', () => {
        expect(arw).toEqual('arw_test')
    })

    it('should match arw_click_test', () => {
        expect(arw_click).toEqual('arw_click_test')
    })

    it('should match cont_par_test', () => {
        expect(cont_par).toEqual('cont_par_test')
    })

    it('should match cont_hr_test', () => {
        expect(cont_hr).toEqual('cont_hr_test')
    })

})