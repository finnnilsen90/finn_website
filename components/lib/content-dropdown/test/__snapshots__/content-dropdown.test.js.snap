// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Content /> mount should match props 1`] = `
<div
  className="content-dropdown_Container"
>
  <h3
    className="title"
  >
    test
  </h3>
  <div
    className="arrow arw_test"
    onClick={[Function]}
  />
  <div
    className="arrow_click arw_click_test"
    onClick={[Function]}
  />
  <hr
    className="content_hr cont_hr_test"
  />
  <div
    className="content_parent cont_par_test"
  >
    <p>
      test content
    </p>
  </div>
</div>
`;

exports[`<Content /> render renders components 1`] = `
<div
  class="content-dropdown_Container"
>
  <h3
    class="title"
  />
  <div
    class="arrow arw_undefined"
  />
  <div
    class="arrow_click arw_click_undefined"
  />
  <hr
    class="content_hr cont_hr_undefined"
  />
  <div
    class="content_parent cont_par_undefined"
  >
    <p />
  </div>
</div>
`;

exports[`<Content /> shallow should match props 1`] = `
<div
  className="content-dropdown_Container"
>
  <h3
    className="title"
  >
    test
  </h3>
  <div
    className="arrow arw_test"
    onClick={[Function]}
  />
  <div
    className="arrow_click arw_click_test"
    onClick={[Function]}
  />
  <hr
    className="content_hr cont_hr_test"
  />
  <div
    className="content_parent cont_par_test"
  >
    <p>
      test content
    </p>
  </div>
</div>
`;
